wrapper/minecraft/src/main/kotlin/org/quiltmc/qkl/wrapper/minecraft/text/StyleDsl.kt

```diff
@@ -0,0 +1,296 @@
+/*
+ * Copyright 2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+@file:Suppress("MemberVisibilityCanBePrivate")
+
+package org.quiltmc.qkl.wrapper.minecraft.text
+
+import net.minecraft.text.ClickEvent
+import net.minecraft.text.HoverEvent
+import net.minecraft.text.MutableText
+import net.minecraft.text.Style
+import net.minecraft.util.Identifier
+import org.quiltmc.qkl.wrapper.minecraft.*
+import org.quiltmc.qkl.wrapper.minecraft.text.mixin.StyleAccessor
+
+/**
+ * Marks objects as being part of QKL's Style Builder DSL.
+ */
+@DslMarker
+public annotation class StyleDsl
+
+/**
+ * A mutable style object that transforms to Minecraft [styles][Style].
+ *
+ * @author NoComment1105
+ */
+@StyleDsl
+public class StyleBuilder {
+    /** A [Color] to apply to the text. */
+    public var color: Color? = null
+        set(value) {
+            if (field != value) {
+                cachedStyle = null
+            }
+            field = value
+        }
+
+    /** Whether to obfuscate the text or not. */
+    public var obfuscated: Boolean? = null
+        set(value) {
+            if (field != value) {
+                cachedStyle = null
+            }
+            field = value
+        }
+
+    /** Whether to format the text in bold or not. */
+    public var bold: Boolean? = null
+        set(value) {
+            if (field != value) {
+                cachedStyle = null
+            }
+            field = value
+        }
+
+    /** Whether to format the text in italics or not. */
+    public var italic: Boolean? = null
+        set(value) {
+            if (field != value) {
+                cachedStyle = null
+            }
+            field = value
+        }
+
+    /** Whether to format the text with a strikethrough or not. */
+    public var strikethrough: Boolean? = null
+        set(value) {
+            if (field != value) {
+                cachedStyle = null
+            }
+            field = value
+        }
+
+    /** Whether to format the text with an underline or not. */
+    public var underlined: Boolean? = null
+        set(value) {
+            if (field != value) {
+                cachedStyle = null
+            }
+            field = value
+        }
+
+    /** A [HoverEvent] to apply to the text. */
```

Comment:
Why are only some of these doc comments inline? Can we standardize them?

Reply:
Property KDoc is inline, except insertion by the look of it. Should I change insertion or make all of them multi-line?